src/main/visual_objs.c

```diff
@@ -534,11 +713,113 @@ void func_800B2880(struct VisualObj* arg0)
     func_8002B318(arg0, 0x18, 0x30);
 }
 
-INCLUDE_ASM("asm/us/main/nonmatchings/visual_objs", func_800B28CC);
+void func_800B28CC(struct VisualObj* arg0)
+{
+    struct VisualObj* temp_s1;
+
+    temp_s1 = arg0->unk50;
+    if ((temp_s1 == 0) || (temp_s1->base.active == 0)) {
+        ZeroObjectState(arg0);
+        return;
+    }
+
+    if (arg0->base.state == 0) {
+        arg0->base.on_screen = 1;
+        arg0->unk38 = temp_s1->unk38;
+        arg0->unk3C = temp_s1->unk3C;
+        arg0->unk30 = temp_s1->unk30;
+        arg0->unk40 = D_8010A588[arg0->base.unk2];
+        arg0->unk42 = temp_s1->unk42;
+        arg0->base.unk16 = D_8010A590[arg0->base.unk2];
+        arg0->base.unk15 = temp_s1->base.unk15;
+        arg0->unk49 = 1;
+        func_80015D60(arg0, D_8010A594[arg0->base.unk2]);
+        arg0->base.state++;
+    } else {
+        func_80015DC8(arg0);
+    }
+
+    arg0->base.x_pos.val = temp_s1->base.x_pos.val;
+    arg0->base.y_pos.val = temp_s1->base.y_pos.val;
+
+    if (D_8010A598[arg0->base.unk2] != 0) {
+        decompress_player_gfx(arg0, 0x140, D_8010A598[arg0->base.unk2]);
+    }
+
+    func_8002B318(arg0, D_8010A5A0[arg0->base.unk2 * 2], D_8010A5A2[arg0->base.unk2 * 2]);
+}
+
+void func_800B2A3C(struct VisualObj* arg0)
+{
+    if (arg0->base.unk2 == 0) {
+        arg0->unk38 = *(u8**)0x1F800014 + (*(u32**)0x1F800014)[0x18 / 4];
+    } else {
+        arg0->unk38 = *(u8**)0x1F800014 + (*(u32**)0x1F800014)[0x24 / 4];
+    }
+
+    arg0->unk40 = 0x520;
+    arg0->unk48 = -1;
+    arg0->unk47 = 0;
+
+    decompress_player_gfx(arg0, 0x140, 0x20);
+    ZeroObjectState(arg0);
+}
+
+void func_800B2AD0(struct VisualObj* arg0)
+{
+    struct PlayerObj* player = &g_Player;
+    s8 var_s1;
+    u8 temp_v1_2;
 
-INCLUDE_ASM("asm/us/main/nonmatchings/visual_objs", func_800B2A3C);
+    if (arg0->base.state == 0) {
+        arg0->base.on_screen = 1;
+        arg0->base.unk2 = -1;
+        arg0->unk3C = *(u8**)0x1F80001C + (*(u32**)0x1F80001C)[0x30 / 4];
+        arg0->unk30 = &D_8011BF40;
+        arg0->unk42 = 0x7802;
+        arg0->unk40 = 0;
+        arg0->base.unk16 = 3;
+        arg0->unk5C = 0;
+        arg0->base.state++;
+    }
+    temp_v1_2 = player->cur_anim;
+    var_s1 = -((((temp_v1_2 + 0x1B) & 0xFF) < 2U) ^ 1);
+    if (((temp_v1_2 + 0x23) & 0xFF) < 2U) {
```

Comment:
These ought to be something like
```
    if(temp_v1_2 > -3 && temp_v1_2 < 0x21) {
```
But I'm slightly off and the numbers are swapped.
https://decomp.me/scratch/nKL14

Reply:
I tried a `switch` statement but no luck. Best I was able to find:

```c
    anim = player->cur_anim;
    var_s1 = -((anim == 0xE5 || anim == 0xE6) ^ 1);
    if (anim == 0xDD || anim == 0xDE) {
        var_s1 = 1;
    }
    if (anim == 0xDF || anim == 0xE0) {
        var_s1 = 2;
    }
    if (anim == 0xE1 || anim == 0xE2) {
        var_s1 = 3;
    }
    if (anim == 0xE3 || anim == 0xE4) {
        var_s1 = 4;
    }
```

The `^ 1` is probably fake but I couldn't get a ternary to match

Reply:
Nice, looks good!